### DISCUSSION TITLE
Conditionally running or bypassing other filters using Sieve scripts conditions

### DISCUSSION THREAD

**Discussion by pacien** [stalwartlabs/smtp-server#28] [Q&A] [answered]
Hi,

According to the documentation, it is possible to filter emails using Sieve
scripts rules and some DNSBLs for example.

However, I could not find how to glue both together in such a way that some
particular DNSBL is queried based on some condition in a Sieve script.

For example, I'd like to do the following:
- always check the sender IP against a known spammer DNSBL, and
- only check the sender IP against a residential IP DNSBL only when there is no
  SPF record nor DKIM signature.

Is there a way to do that?

**Comment by mdecimus** [comment, marked as answer]
At the moment it is not possible to query DNSBL from Sieve filter but this functionality is going to be added on the next release (in 1 or 2 months approx).  Currently you can [set DNSBL](https://stalw.art/docs/smtp/filter/dnsbl) in the configuration file but they will be always checked.

> only check the sender IP against a residential IP DNSBL only when there is no SPF record nor DKIM signature.

Be aware that many spammers nowadays have valid DKIM and SPF records.
